### PR TITLE
EWS - fix debug messages & add commands

### PR DESCRIPTION
**Pull request by erezh31**
1. add debug log when exception and informative error messages
2. add create folder command
3. add mark as junk command
4. improve integration test

### CODE REVIEW

**Comment by erezh31** [comment]
fixes:
https://github.com/demisto/etc/issues/11081

File: Integrations/integration-EWSMailSender.yml [hunk 1]
```diff
@@ -42,34 +43,46 @@ configuration:
   defaultvalue: "false"
   type: 8
   required: false
+- display: Has impersonation rights.
+  name: impersonation
```
Comment (aviadl) [inline Integrations/integration-EWSMailSender.yml]: why do we need impersonation for send ?

File: Integrations/integration-EWSMailSender.yml [hunk 2]
```diff
@@ -42,34 +43,46 @@ configuration:
...
     VERSION_STR = demisto.params()['defaultServerVersion']
     EWS_SERVER = demisto.params()['ewsServer']
     USERNAME = demisto.params()['credentials']['identifier']
-    ACCOUNT_EMAIL = USERNAME
+    ACCOUNT_EMAIL = demisto.params().get('mailbox', None)
+    if not ACCOUNT_EMAIL:
+        if "@" in USERNAME:
+            ACCOUNT_EMAIL = USERNAME
+    if ACCOUNT_EMAIL is None:
+        raise Exception("Please provide a valid email address in the mailbox field")
```
Comment (aviadl) [inline Integrations/integration-EWSMailSender.yml]: Remove the `Please`

File: Integrations/integration-EWSMailSender.yml [hunk 3]
```diff
@@ -197,17 +210,36 @@ script:
 
 
     def test_module():
+        BaseProtocol.TIMEOUT = 15
```
Comment (aviadl) [inline Integrations/integration-EWSMailSender.yml]: In what scale is this timeout?

File: Integrations/integration-EWSMailSender.yml [hunk 4]
```diff
@@ -197,17 +210,36 @@ script:
...
+
+    except Exception, e:
+        import time
+        time.sleep(2)
+        log_stream.seek(0)
+        debug_log = log_stream.read()
+        error_message = ""
+        if "Status code: 401" in debug_log:
+            error_message = "Got unauthorized from the server. " \
+                                 "Please check credentials are correct and authentication method are supported. " \
```
Comment (aviadl) [inline Integrations/integration-EWSMailSender.yml]: Remove `Please`

File: Integrations/integration-EWSMailSender.yml [hunk 5]
```diff
@@ -197,17 +210,36 @@ script:
...
+                                 "If you are using NTLM, you can try using 'domain\\username' as username for authentication"
```
Comment (aviadl) [inline Integrations/integration-EWSMailSender.yml]: We know if he uses NTLM no?

File: Integrations/integration-EWSMailSender.yml [hunk 6]
```diff
@@ -197,17 +210,36 @@ script:
...
     config = prepare()
     args = prepare_args(demisto.args())
 
-    if demisto.command() == 'test-module':
-        test_module()
-    elif demisto.command() == 'send-mail':
-        demisto.results(send_email(**args))
+    try:
+        if demisto.command() == 'test-module':
+            test_module()
```
Comment (aviadl) [inline Integrations/integration-EWSMailSender.yml]: Do you want to actually send an email in test? 
Will make the test more reliable

File: Integrations/integration-EWSMailSender.yml (hunk 1 above)
Comment (erezh31) [reply Integrations/integration-EWSMailSender.yml]: Sometimes only impersonation will be possible (in some exchange config)

File: Integrations/integration-EWSMailSender.yml (hunk 2 above)
Comment (erezh31) [reply Integrations/integration-EWSMailSender.yml]: k

File: Integrations/integration-EWSMailSender.yml (hunk 3 above)
Comment (erezh31) [reply Integrations/integration-EWSMailSender.yml]: It's only when test (for authentication only). I think it's enough. I can increase it if you think it's too low

File: Integrations/integration-EWSMailSender.yml (hunk 4 above)
Comment (erezh31) [reply Integrations/integration-EWSMailSender.yml]: k

File: Integrations/integration-EWSMailSender.yml (hunk 5 above)
Comment (erezh31) [reply Integrations/integration-EWSMailSender.yml]: yes

File: Integrations/integration-EWSMailSender.yml (hunk 6 above)
Comment (erezh31) [reply Integrations/integration-EWSMailSender.yml]: Do you mean send a mail to the same mailbox?
don't know.. I think it's weird to send mail when test

**Pull request merged**